File: lib/models/timeslot.dart
class TimeSlot {
  final int? id;
  final int subCourtId;
  final String date;
  final String startTime;
  final String endTime;
  final bool isBooked;

  TimeSlot({
    this.id,
    required this.subCourtId,
    required this.date,
    required this.startTime,
    required this.endTime,
    this.isBooked = false,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'subCourtId': subCourtId,
      'date': date,
      'startTime': startTime,
      'endTime': endTime,
      'isBooked': isBooked,
    };
  }

  factory TimeSlot.fromJson(Map<String, dynamic> json) {
    return TimeSlot(
      id: json['id'],
      subCourtId: json['subCourtId'],
      date: json['date'] ?? '',
      startTime: json['startTime'],
      endTime: json['endTime'],
      isBooked: json['isBooked'] ?? false,
    );
  }
}

File: backend/BadmintonBookingAPI/Models/TimeSlot.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace BadmintonBookingAPI.Models;

public class TimeSlot
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int SubCourtId { get; set; }

    [Required]
    [MaxLength(10)]
    public string Date { get; set; } = string.Empty;

    [Required]
    [MaxLength(10)]
    public string StartTime { get; set; } = string.Empty;

    [Required]
    [MaxLength(10)]
    public string EndTime { get; set; } = string.Empty;

    public bool IsBooked { get; set; } = false;

    [ForeignKey("SubCourtId")]
    [JsonIgnore]
    public SubCourt? SubCourt { get; set; }

    [JsonIgnore]
    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}
